pkg/controller/factory/base_controller.go

```diff
@@ -120,35 +121,37 @@ func (c *baseController) Run(ctx context.Context, workers int) {
 	// if scheduled run is requested, run the cron scheduler
 	if c.resyncSchedules != nil {
 		scheduler := cron.New()
+		scheduledJob := newScheduledJob(c.Name(), c.syncContext.Queue())
 		for _, s := range c.resyncSchedules {
-			scheduler.Schedule(s, newScheduledJob(c.name, c.syncContext.Queue()))
+			schedule, err := cron.ParseStandard(s)
+			if err != nil {
+				panic(fmt.Errorf("invalid controller schedule %q: %v", s, err))
+			}
+			klog.Infof("Controller %s will resync on %s", c.Name(), s)
+			scheduler.Schedule(schedule, scheduledJob)
 		}
 		scheduler.Start()
 		defer scheduler.Stop()
-	}
-
-	// runPeriodicalResync is independent from queue
-	if c.resyncEvery > 0 {
-		workerWg.Add(1)
-		go func() {
-			defer workerWg.Done()
-			c.runPeriodicalResync(ctx, c.resyncEvery)
-		}()
+		// always schedule initial run, regardless of the schedule
+		go scheduledJob.Run()
```

Comment:
why not queue directly?

Reply:
scheduledJob.Run() does queue, I found this more cleaner for the intent